atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/atomix/InvalidatingLeaderProxyTest.java

```diff
@@ -98,6 +117,65 @@ public void shouldResetDelegateOnLeaderChange() {
     }
 
     @Test
+    public void shouldCloseDelegateIfCloseableAndNotLeader() throws IOException {
+        ByteChannel mockedResource = mock(ByteChannel.class);
+        ByteChannel proxiedResource = InvalidatingLeaderProxy.create(
+                LOCAL_MEMBER,
+                LEADER_ID,
+                election,
+                () -> mockedResource,
+                ByteChannel.class);
+
+        setLeader(LOCAL_MEMBER_ID);
+        proxiedResource.isOpen();
+
+        setLeaderWithoutCallbacks(null);
+        assertThatThrownBy(proxiedResource::isOpen).isInstanceOf(ServiceUnavailableException.class);
+
+        verify(mockedResource, times(1)).close();
+    }
+
+    @Test
+    public void shouldCloseDelegateIfCloseableAndTermChanges() throws IOException {
```

Comment:
nit: this should be `...AndLeaderChanges()` to be strictly consistent with the other tests here; `...TermChanges()` has been used to test the case where we stay as the leader through two elections.